Source/Scene/PointPrimitiveCollection.js

```diff
@@ -0,0 +1,976 @@
+
+                this._spPick = context.replaceShaderProgram(this._spPick, vs, fs, attributeLocations);
+                this._compiledShaderScaleByDistancePick = this._shaderScaleByDistance;
+                this._compiledShaderTranslucencyByDistancePick = this._shaderTranslucencyByDistance;
+            }
+
+            va = this._vaf.va;
+            vaLength = va.length;
+
+            pickList.length = vaLength;
+            for (j = 0; j < vaLength; ++j) {
+                command = pickList[j];
+                if (!defined(command)) {
+                    command = pickList[j] = new DrawCommand({
+                        primitiveType : PrimitiveType.POINTS,
+                        pass : Pass.OPAQUE,
+                        owner : this
+                    });
+                }
+
+                command.boundingVolume = boundingVolume;
+                command.modelMatrix = modelMatrix;
+                command.shaderProgram = this._spPick;
+                command.uniformMap = this._uniforms;
+                command.vertexArray = va[j].va;
+                command.renderState = this._rs;
+
+                commandList.push(command);
+            }
+        }
+    };
+
+    /**
+     * Returns true if this object was destroyed; otherwise, false.
+     * <br /><br />
+     * If this object was destroyed, it should not be used; calling any function other than
+     * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.
+     *
+     * @returns {Boolean} <code>true</code> if this object was destroyed; otherwise, <code>false</code>.
+     *
+     * @see PointPrimitiveCollection#destroy
+     */
+    PointPrimitiveCollection.prototype.isDestroyed = function() {
+        return false;
+    };
+
+    /**
+     * Destroys the WebGL resources held by this object.  Destroying an object allows for deterministic
+     * release of WebGL resources, instead of relying on the garbage collector to destroy this object.
+     * <br /><br />
+     * Once an object is destroyed, it should not be used; calling any function other than
+     * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.  Therefore,
+     * assign the return value (<code>undefined</code>) to the object as done in the example.
+     *
+     * @returns {undefined}
+     *
+     * @exception {DeveloperError} This object was destroyed, i.e., destroy() was called.
+     *
+     * @see PointPrimitiveCollection#isDestroyed
+     *
+     * @example
+     * pointPrimitives = pointPrimitives && pointPrimitives.destroy();
+     */
+    PointPrimitiveCollection.prototype.destroy = function() {
+        this._sp = this._sp && this._sp.destroy();
+        this._spPick = this._spPick && this._spPick.destroy();
+        this._vaf = this._vaf && this._vaf.destroy();
+        this._destroyPointPrimitives();
+
+        return destroyObject(this);
+    };
+
+    PointPrimitiveCollection.prototype._destroyPointPrimitives = function() {
```

Comment:
We could use a plain file-level scope function and pass in `this`.  Same for `BillboardCollection`.  This code is old...